packages/phoenix-client/src/phoenix/client/resources/experiments/evaluators.py

```diff
@@ -65,8 +124,12 @@ def _bind_evaluator_signature(sig: inspect.Signature, **kwargs: Any) -> inspect.
     )
 
 
-def _default_eval_scorer(result: Any) -> EvaluationResult:
+def _default_eval_scorer(result: Any) -> Union[EvaluationResult, "ScoreResult"]:
```

Comment:
why the union type here. Doesn't this just make a cascade of type checking downstream? Shouldn't you normalize the output here?

Reply:
The main reason is that having the type checking downstream allows us to use the additional information contained in the `Score` object when it's finally used to submit an EvaluationResult payload to the API. If the type is normalized here, we lose information (like the evaluator name that's specified on the score) that would otherwise override the inferred evaluator name. This is especially important when we expand support for multiple score outputs.

The implementation that @ehutt and I dogfooded originally actually *did* normalize the scores at this point, but we noted a lot of inconsistent behavior with evaluators that returned scores